src/app/player-creation/player-creation-modal/player-creation-modal.component.spec.ts

```diff
@@ -1,20 +1,24 @@
 import { PlayerCreationModalComponent } from './player-creation-modal.component';
-import { FormBuilder } from '@angular/forms';
+import { FormBuilder, FormGroup } from '@angular/forms';
 import { MatDialogRef } from '@angular/material/dialog';
 
 describe('PlayerCreationModalComponent', () => {
   let component: PlayerCreationModalComponent;
-
   let formBuilder: FormBuilder;
   let dialogRef: MatDialogRef<PlayerCreationModalComponent>;
 
   beforeEach(() => {
     formBuilder = { group: (_: any) => {} } as FormBuilder;
-
+    dialogRef = { close: _ => {} } as MatDialogRef<PlayerCreationModalComponent>;
     component = new PlayerCreationModalComponent(formBuilder, dialogRef);
   });
 
-  it('should create', () => {
-    expect(component).toBeTruthy();
+  it('sends the name when closing the modal', () => {
+    component.formGroup = { value: { name: 'Imene' } } as FormGroup;
+    spyOn(dialogRef, 'close');
+
+    component.create();
+
+    expect(dialogRef.close).toHaveBeenCalledWith('Imene');
```

Comment:
ImenEEE please!

Reply:
No, we are not dealing with jokes here imenE. This is a serious project.